lib/Compiler.js

```diff
@@ -2,520 +2,524 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var path = require("path");
-var Tapable = require("tapable");
-
-var Compilation = require("./Compilation");
-var Stats = require("./Stats");
-var NormalModuleFactory = require("./NormalModuleFactory");
-var ContextModuleFactory = require("./ContextModuleFactory");
-
-var makePathsRelative = require("./util/identifier").makePathsRelative;
-
-function Watching(compiler, watchOptions, handler) {
-	this.startTime = null;
-	this.invalid = false;
-	this.handler = handler;
-	this.closed = false;
-	if(typeof watchOptions === "number") {
-		this.watchOptions = {
-			aggregateTimeout: watchOptions
-		};
-	} else if(watchOptions && typeof watchOptions === "object") {
-		this.watchOptions = Object.assign({}, watchOptions);
-	} else {
-		this.watchOptions = {};
+"use strict";
+
+const path = require("path");
+const Tapable = require("tapable");
+
+const Compilation = require("./Compilation");
+const Stats = require("./Stats");
+const NormalModuleFactory = require("./NormalModuleFactory");
+const ContextModuleFactory = require("./ContextModuleFactory");
+
+const makePathsRelative = require("./util/identifier").makePathsRelative;
+
+class Watching {
+	constructor(compiler, watchOptions, handler) {
+		this.startTime = null;
+		this.invalid = false;
+		this.handler = handler;
+		this.closed = false;
+		if(typeof watchOptions === "number") {
+			this.watchOptions = {
+				aggregateTimeout: watchOptions
+			};
+		} else if(watchOptions && typeof watchOptions === "object") {
+			this.watchOptions = Object.assign({}, watchOptions);
+		} else {
+			this.watchOptions = {};
+		}
+		this.watchOptions.aggregateTimeout = this.watchOptions.aggregateTimeout || 200;
+		this.compiler = compiler;
+		this.running = true;
+		this.compiler.readRecords(err => {
+			if(err) return this._done(err);
+
+			this._go();
+		});
 	}
-	this.watchOptions.aggregateTimeout = this.watchOptions.aggregateTimeout || 200;
-	this.compiler = compiler;
-	this.running = true;
-	this.compiler.readRecords(function(err) {
-		if(err) return this._done(err);
-
-		this._go();
-	}.bind(this));
-}
 
-Watching.prototype._go = function() {
-	var self = this;
-	self.startTime = Date.now();
-	self.running = true;
-	self.invalid = false;
-	self.compiler.applyPluginsAsync("watch-run", self, function(err) {
-		if(err) return self._done(err);
-		self.compiler.compile(function onCompiled(err, compilation) {
+	_go() {
+		const self = this;
```

Comment:
converted 'self' to 'this' in the _go method. Shifted some normal functions around and converted them to arrow functions.